gems/sorbet-runtime/lib/types/private/methods/signature.rb

```diff
@@ -82,8 +83,13 @@ def initialize(method:, method_name:, raw_arg_types:, raw_return_type:, bind:, m
     if parameters.size != raw_arg_types.size
       raise "The declaration for `#{method.name}` has arguments with duplicate names"
     end
+    params_size = parameters.size
+    i = 0
+    while i < params_size
```

Comment:
This is a completely minor thing but I always liked the pattern of structuring this like
```suggestion
    i = -1
    while i < params_size
      i += 1
```

So that all the index logic happens in one place.

Reply:
The proposed change is not the common pattern for sorbet-runtime, and it also has an edge case when `params_size` is 0.  I'm opting to leave this as is, though I do agree that for this particular function, having some 50 lines between test and increment is suboptimal.

Reply:
Oh right I forgot that i usually use this with for auto loops